### ISSUE TITLE
vm error when trying to play nehahra

### ISSUE THREAD

**Issue by revelator** [j0zzz/JoeQuake#80]
Just in case you decide to look at it again :) getting an pf_ function error when trying to play nehara, strangely the demos work.

**Comment by Flecked42** [comment]
Looks like the error only happens when loading neh1m1, but i never really tested the full mod.

![joequake000](https://user-images.githubusercontent.com/90434262/204963079-6f7e406b-8bb9-4ece-aa8a-4def5b4eab51.png)

**Comment by revelator** [comment]
aye seems only the first map is affected, tracing it leads to PR_findstring via one of the edict macros not been able to trace it further as im not home atm. Might just be a small glitch i reckon, i changed the Host_Error call to a Con_DPrintf call for the time being and it plays the rest with no problems.

**Comment by j0zzz** [comment]
Started checking this a few days ago. I've merged the code from QuakeSpasm more than 2 years ago which reads data from the progs.dat in a more optimal way in terms of memory management of strings. But seems like it does something in a wrong way in case of the nehahra progs. Gonna continue investigating it later as I'm busy with another project currently.

**Comment by revelator** [comment]
well nehahra was actually built before any of the quake code was public and the qc code was compiled using a hacked together qc compiler with some oddities :) i actually think the first qc compiler able to handle the odd format was spikes ?.

**Comment by revelator** [comment]
well besides the tools they built themself that is ;).

**Comment by revelator** [comment]
yep found the post on func_msgboard the old nehahra qc compiler was done in pascal and called reacc.

**Comment by Flecked42** [comment]
Nehahra compatibility seems to be a long cause, I don't think any modern engines actually support it.

**Comment by j0zzz** [comment]
Yeah well I have the original source code of it (QC and EXE) from long ago when I added support for it into JoeQuake. Everything is supported, it's this particular QuakeC issue that causes the problem right now.

**Comment by revelator** [comment]
indeed :) i have a hunch it might be a minor problem though.

**Comment by revelator** [comment]
aye thats the one

**Comment by revelator** [comment]
ok tried as admin but no dice, the save path is wrong it seems it tries to save to C:\quake\-nehara\sav0.so Oo which is a rather strange format as .so is a unix shared object format normally lol. not sure why it prefixes the nehahra folder with a - either.

**Comment by revelator** [comment]
ah wait a sec, quakespasm yeah i thought i seen this code somewhere :) quakespasm has problems with nehahra, it can play some maps but there are some bugs with it i heard.
